core/src/main/java/org/apache/calcite/adapter/java/ReflectiveSchema.java

```diff
@@ -235,6 +240,13 @@ private static Enumerable toEnumerable(final Object o) {
         "Cannot convert " + o.getClass() + " into a Enumerable");
   }
 
+  private static @Nullable Double toRowCount(final Object o) {
+    if (o.getClass().isArray()) {
```

Comment:
Would it make sense to consider also the case where `o` is a Collection?

Reply:
Most likely, but I don't have a concrete use case for that variation so I have just implemented the one I have myself and expected anyone with additional cases could follow this up with a separate PR.

Reply:
Completed in #3634 .